Tests/IntegrationTests.Shared/ObjectIntegrationTests.cs

```diff
@@ -247,6 +247,23 @@ public void IteratePeople()
             Assert.That (iterCount, Is.EqualTo (3));
         }
 
+        [Test]
+        public void IsValidReturnsTrueWhenObjectRowIsAttached()
+        {
+            _realm.Write(() =>
+            {
+                var p1 = _realm.CreateObject<Person>();
+                Assert.That(p1.IsValid);
+
+                _realm.Remove(p1);
+                Assert.That(p1.IsValid, Is.False);
+            });
+
+            // IsValid should always return false for standalone objects
```

Comment:
It doesn't match the semantics of Cocoa's equivalent either. `-[RLMObject isInvalidated]` returns true only if the object is managed and its row is detached. An unmanaged object is always valid.